src/AbstractTrackingListener.php

```diff
@@ -225,7 +226,18 @@ protected function updateField($object, $eventAdapter, $meta, $field)
             }
         }
 
-        $property->setValue($object, $newValue);
+        if (!empty($config['setterMethod'][$field])) {
+            $reflectionClass = $meta->getReflectionClass();
+            $setterName = $config['setterMethod'][$field];
+
+            if (!$reflectionClass->hasMethod($setterName)) {
+                throw new InvalidMappingException("Setter method - [{$setterName}] does not exist in class - {$meta->getName()}");
+            }
+
+            $reflectionClass->getMethod($setterName)->invoke($object, $newValue);
```

Comment:
Is there a reason for using reflection to check and call the method here instead of `$object->{$setterName}($newValue)`?  Same for the `SoftDeleteableListener`.

Reply:
Calling $object->{$setterName}($newValue) without checking if the method exists would throw a warning if it doesn't. I do prefer an Exception with proper information.

The only reason for using reflection to actually call the method is because it's already loaded for the check.

Reply:
The check can be done with `method_exists()` as well.

I think the only gotcha here is that this could unintentionally call into a non-public setter, whereas `method_exists()` and the direct call restricts it to public scope (unless you check that through reflection as well).

Reply:
Yes, of course. I'll replace the reflection checks.

Reply:
I was also considering to auto-detect if there is a setter method for the configured field. That would be a bc-breaking change though.

Reply:
Yeah, at this point I think opt-in with config is better than a breaking behavioral change.